internal/host/host.go

```diff
@@ -139,7 +144,7 @@ type API interface {
 	EnableHost(ctx context.Context, h *models.Host, db *gorm.DB) error
 	// Install host - db is optional, for transactions
 	Install(ctx context.Context, h *models.Host, db *gorm.DB) error
-	GetStagesByRole(role models.HostRole, isbootstrap bool) []models.HostStage
+	GetStagesByRole(role models.HostRole, isbootstrap bool, isSNO bool) []models.HostStage
```

Comment:
Why do you pass `isSNO` in to the function if you can use `hostutil.IsSingleNode()` ?

Reply:
IsSingleNode util queries the database for the cluster which is redundant in many cases were the cluster is already at hand and in other cases we should take care not to ask for a cluster where the host is unbound.